src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -58,22 +61,53 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
         //Declare and initiate lists to update
         List<Contact> contactsToUpdate = new List<Contact>();
         List<Grade_Enrollment__c> gradeEnrolsToUpdate = new List<Grade_Enrollment__c>();
+        
+        //Declare and initiate oldMap
+        Map<ID, Grade_Enrollment__c> oldMap = new Map<ID, Grade_Enrollment__c>();
+        if (listOld != NULL) {
+            oldMap = new Map<ID, Grade_Enrollment__c>((List<Grade_Enrollment__c>)listOld);
+        }
+        //Declare new map to capture Contacts and their related Grade Enrollments
+        Map<ID, List<Grade_Enrollment__c>> contactToGradeEnroll = new Map<ID, List<Grade_Enrollment__c>>();
+             
+        //Declare and initiate set of Grade Enrollment Ids that were changed from a stauts of 'Current'
+        Set<Id> formerGradeEnrolIds = new Set<Id>();
+        
+        //Declare and initiate set of related Contact Ids that had its Grade Enrollments changed from a stauts of 'Current'
+        Set<Id> contactsIdsFormer = new Set<Id>(); 
+            
+        //Declare and initiate lists to show errors on Grade Enrollment records
+        List<Grade_Enrollment__c> gradeEnrolsErrorRecords = new List<Grade_Enrollment__c>();   
+         
+        // Declare and initiate map of Contact Id to its Current Grade Level
+        Map<Id, String> contactIdToCurrentGradeLevel = new Map<Id, String>(); 
 
         //After INSERT and After UPDATE
         if(triggerAction == hed.TDTM_Runnable.Action.AfterInsert 
             || triggerAction == hed.TDTM_Runnable.Action.AfterUpdate) {
             if (listNew != null && listNew.size() > 0) {
                 for (Integer i = 0; i < listNew.size(); i++) {
-                    Grade_Enrollment__c gradeErol = (Grade_Enrollment__c)listNew[i];
-                    if (gradeErol.Status__c != null 
-                        && gradeErol.Status__c == 'Current') {
+                    Grade_Enrollment__c gradeEnrol = (Grade_Enrollment__c)listNew[i];
+                    if (gradeEnrol.Status__c != NULL 
+                        && gradeEnrol.Status__c == 'Current') {
                         //Collect Ids of related contacts
-                        relatedContactIds.add(gradeErol.Contact__c);
-                        currentGradeEnrolIds.add(gradeErol.Id);
+                        relatedContactIds.add(gradeEnrol.Contact__c);
+                        currentGradeEnrolIds.add(gradeEnrol.Id);
+                    } else if (triggerAction == hed.TDTM_Runnable.Action.AfterUpdate
+                        && oldMap.get(gradeEnrol.ID).Status__c == 'Current'
+                        && gradeEnrol.Status__c != 'Current') {
```

Comment:
This condition is already covered in else if.

Reply:
Hi @xinnanlisf Sry I am not sure what this meant? Can you please clarify? Thx.